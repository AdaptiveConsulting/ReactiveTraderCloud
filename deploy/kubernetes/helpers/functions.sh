#! /bin/bash

listcontains() {
  for word in $1; do
    [[ $word = $2 ]] && return 0
  done
  return 1
}
