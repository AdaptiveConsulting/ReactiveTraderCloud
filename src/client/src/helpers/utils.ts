import d3 from 'd3'

const numberConvertRegex = /^([0-9\.]+)?([MK]{1})?$/

/**
 * Returns the expanded price from k/m shorthand.
 */
export function convertNotionalShorthandToNumericValue(notionalShorthandRaw: string): number {
  const notionalShorthand = String(notionalShorthandRaw).toUpperCase().replace(/,/g, '')
  const matches = notionalShorthand.match(numberConvertRegex)

  let notionalShorthandNum = 0
  
  if (notionalShorthand.length && matches && matches.length) {
    notionalShorthandNum = Number(matches[1]) ? Number(matches[1]) : 1
    if (matches[2]) {
      notionalShorthandNum = notionalShorthandNum * (matches[2] === 'K' ? 1000 : 1000000)
    }
  }

  return notionalShorthandNum
}

export function hasShorthandInput(notionalShorthandRaw: string): boolean | null {
  const notionalShorthand = String(notionalShorthandRaw).toUpperCase().replace(/,/g, '')
  const matches = notionalShorthand.match(numberConvertRegex)
  return matches && matches[2] !== undefined // found K or M
}

export function formatDate(date: string, format: string = '%b %e, %H:%M:%S'): string {
  return d3.time.format(format)(date)
}
