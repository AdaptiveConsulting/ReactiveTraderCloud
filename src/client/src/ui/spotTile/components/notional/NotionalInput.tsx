import numeral from 'numeral'
import React, { PureComponent } from 'react'
import { NUMERAL_FORMAT, isEditMode } from '../TileBusinessLogic'
import {
  InputWrapper,
  CurrencyPairSymbol,
  Input,
  ResetInputValue,
  MessagePlaceholder,
} from './styled'

const DOT = '.'
const ENTER = 'Enter'
const CHAR_CODE_DOT = 46 // .
const CHAR_CODE_0 = 48 // 0
const CHAR_CODE_9 = 57 // 9
const CHAR_CODE_UNIT_SEP = 31 // US (unit separator)
const SHORTCUT_CHAR_CODES = [75, 77, 107, 109] // K, M, k, m

export type ValidationMessage = null | {
  type: 'warning' | 'error' | 'info'
  content: string
}

export interface NotionalUpdate {
  value: string
  type: string
}

interface Props {
  currencyPairSymbol: string
  notional: string
  updateNotional: (notionalUpdate: NotionalUpdate) => void
  resetNotional: () => void
  validationMessage: ValidationMessage
  showResetButton?: boolean
  disabled: boolean
}

interface State {
  showMessage: boolean
}

export default class NotionalInput extends PureComponent<Props, State> {
  private inputRef = React.createRef<HTMLInputElement>()

  handleFocus = (event: React.FocusEvent<HTMLInputElement>) => {
    event.target.select()
  }

  handleKeyPressNotionalInput = (event: React.KeyboardEvent<HTMLInputElement>) => {
    const { charCode, currentTarget, nativeEvent } = event
    if (event.key === ENTER) {
      this.handleChange(event)
    } else if (charCode === CHAR_CODE_DOT) {
      // only allow one dot unless the existing dot is in the current
      // text selection while replacing existing text.
      const { value, selectionStart, selectionEnd } = currentTarget
      const textWithoutSelection = value.replace(value.substring(selectionStart, selectionEnd), '')
      if (textWithoutSelection.match(/\./g)) {
        nativeEvent.stopImmediatePropagation()
        event.preventDefault()
      }
    } else if (!this.inputIsAllowed(charCode)) {
      nativeEvent.stopImmediatePropagation()
      event.preventDefault()
    }
  }

  handleChange = (event: React.FormEvent<HTMLInputElement>) => {
    const { updateNotional } = this.props
    const { currentTarget, type } = event
    const inputValue = currentTarget.value.trim()
    const value = this.formatNotional(inputValue)
    updateNotional({ value, type })
  }

  handleResetNotional = () => {
    this.props.resetNotional()
  }

  formatNotional = (notional: string) => {
    // user may be trying to enter decimals or
    // user may be deleting previous entry (empty string)
    // in those cases, format and update only when completed.
    if (notional === '.') {
      return '0.'
    }
    const lastTwoChars = notional.substr(-2)
    const shouldFormat = !isEditMode(notional) && lastTwoChars.indexOf(DOT) === -1
    return shouldFormat ? numeral(notional).format(NUMERAL_FORMAT) : notional
  }

  inputIsAllowed = (charCode: number) => {
    // allow charcter codes before the Unit Separator to catch Shift, Backspace, etc
    if (charCode <= CHAR_CODE_UNIT_SEP) {
      return true
    }
    // allow shortcut values
    if (SHORTCUT_CHAR_CODES.indexOf(charCode) !== -1) {
      return true
    }
    // allow numeric values:
    if (charCode >= CHAR_CODE_0 && charCode <= CHAR_CODE_9) {
      return true
    }
    return false
  }

  render() {
    const {
      currencyPairSymbol,
      notional,
      validationMessage,
      showResetButton,
      disabled,
    } = this.props

    return (
      <InputWrapper altLayout={Boolean(validationMessage)}>
        <CurrencyPairSymbol>{currencyPairSymbol}</CurrencyPairSymbol>
        <Input
          type="text"
          ref={this.inputRef}
          value={notional}
          onFocus={this.handleFocus}
          onChange={this.handleChange}
          onBlur={this.handleChange}
          onKeyPress={this.handleKeyPressNotionalInput}
          validationMessage={validationMessage}
          disabled={disabled}
        />
        {showResetButton && (
          <ResetInputValue onClick={this.handleResetNotional}>
            <i className="fas fa-redo fa-flip-horizontal" />
          </ResetInputValue>
        )}
        {validationMessage && (
          <MessagePlaceholder validationMessageType={validationMessage.type}>
            {validationMessage.content}
          </MessagePlaceholder>
        )}
      </InputWrapper>
    )
  }
}
