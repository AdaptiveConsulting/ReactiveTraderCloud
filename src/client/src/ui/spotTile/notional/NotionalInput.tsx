import * as React from 'react'
import * as numeral from 'numeral'
import { utils } from '../../../system'
import * as classnames from 'classnames'
import './notionalInput.scss'

const NUMERAL_FORMAT      = '0,000,000[.]00'
const DOT                 = '.'
const ENTER               = 'Enter'
const CHAR_CODE_DOT       = 46
const CHAR_CODE_0         = 48
const CHAR_CODE_9         = 57
const CHAR_CODE_UNIT_SEP  = 31
const SHORTCUT_CHAR_CODES = [75, 77, 107, 109] // K, M, k, m

const MAX_NOTIONAL_VALUE = 1000000000

export interface CurrencyPair {
  symbol: string
  base: string
}

export interface NotionalInputProps {
  className: string
  notional: number
  currencyPair: CurrencyPair
  onNotionalInputChange: (value: number) => void
}

export default class NotionalInput extends React.Component<NotionalInputProps, {}> {
  props: NotionalInputProps
  refs: any
  shouldComponentUpdate(nextProps: any, nextState: any) {
    return this.props.className !== nextProps.className ||
      this.props.notional !== nextProps.notional ||
      // currencyPair always here
      this.props.currencyPair.symbol !== nextProps.currencyPair.symbol ||
      this.props.onNotionalInputChange !== nextProps.onChange
  }

  render() {
    const { className, currencyPair, notional } = this.props

    const formattedSize = numeral(notional).format(NUMERAL_FORMAT)
    const classes = classnames('notional', className)

    return (
      <div className={classes}>
        <label className="notional__currency-pair">{currencyPair.base}</label>
        <input className="notional__size-input"
          type="text"
          ref="notionalInput"
          defaultValue={formattedSize}
          onClick={this.handleSelect}
          onChange={(e: any) => this.handleInputChange(e)}
          onBlur={(e: any) => this.processNotional(e.target.value)}
          onKeyPress={(e: any) => this.handleKeyPressNotionalInput(e)}/>
      </div>
    )
  }

  handleKeyPressNotionalInput(e: any) {
    const charCode = e.charCode

    if (e.key === ENTER) {
      this.processNotional(e.target.value)
    } else if (charCode === CHAR_CODE_DOT) {
      // only allow one dot
      const numDots = e.target.value.match(/\./g).length
      if (numDots > 0) {
        e.nativeEvent.stopImmediatePropagation()
        e.preventDefault()
      }
    } else if (!this.inputIsAllowed(charCode)) {
      e.nativeEvent.stopImmediatePropagation()
      e.preventDefault()
    }
  }

  processNotional(inputValue: string) {
    const inputValueTrimmed = inputValue.trim()
    let notional: any = utils.convertNotionalShorthandToNumericValue(inputValueTrimmed)
    if (notional >= MAX_NOTIONAL_VALUE) {
      notional = 0
    }
    if (!isNaN(notional)) {
      // send temp notional back to parent
      this.props.onNotionalInputChange(notional)

      // user may be trying to enter decimals. restore BACK into input
      if (inputValueTrimmed.indexOf(DOT) === inputValueTrimmed.length - 1) {
        notional = notional + DOT
      }
      // propagate change back to dom node's value
      this.refs.notionalInput.value = numeral(notional).format(NUMERAL_FORMAT)
    }
  }

  handleSelect(e: any) {
    const el = e.target
    el.setSelectionRange(0, el.value.length)
  }

  handleInputChange(e: any) {
    const rawValue = (this.refs.notionalInput.value || e.target.value).trim()
    // check for a shortcut input
    if (utils.hasShorthandInput(rawValue)) {
      this.processNotional(rawValue)
    }
  }

  inputIsAllowed(charCode: number) {
    // allow charcter codes before the Unit Separator to catch Shift, Backspace, etc
    if (charCode <= CHAR_CODE_UNIT_SEP) return true

    // allow shortcut values
    if (SHORTCUT_CHAR_CODES.indexOf(charCode) !== -1) return true

    // allow numeric values:
    if (charCode >= CHAR_CODE_0 && charCode <= CHAR_CODE_9) return true
    return false
  }
}
