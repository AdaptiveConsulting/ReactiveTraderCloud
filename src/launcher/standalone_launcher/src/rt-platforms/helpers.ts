import { UAParser } from 'ua-parser-js'

export const isRunningInIE = () => {
  const browser = new UAParser().getBrowser().name
  return browser && browser.indexOf('IE') !== -1
}
