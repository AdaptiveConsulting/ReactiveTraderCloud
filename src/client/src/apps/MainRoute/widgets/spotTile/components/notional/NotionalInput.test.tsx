import React from 'react'
import NotionalInput, { ValidationMessage } from './NotionalInput'
import { renderWithTheme } from '../../../../../../../__tests__/helpers'

const notional = '1,000,000'
const currencyPairSymbol = 'EURUSD'
const updateNotional = jest.fn()
const resetNotional = jest.fn()
const inputValidationMessageError: ValidationMessage = {
  type: 'error',
  content: 'Some error',
}
const inputValidationMessageWarning: ValidationMessage = {
  type: 'warning',
  content: 'Some warning',
}
const inputValidationMessageInfo: ValidationMessage = {
  type: 'info',
  content: 'Some information',
}

// Dark theme version
test('NotionalInput default', () => {
  const component = renderWithTheme(
    <NotionalInput
      notional={notional}
      currencyPairSymbol={currencyPairSymbol}
      updateNotional={updateNotional}
      resetNotional={resetNotional}
      validationMessage={null}
      showResetButton={false}
      disabled={false}
    />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('NotionalInput with error', () => {
  const component = renderWithTheme(
    <NotionalInput
      notional={notional}
      currencyPairSymbol={currencyPairSymbol}
      updateNotional={updateNotional}
      resetNotional={resetNotional}
      validationMessage={inputValidationMessageError}
      showResetButton={false}
      disabled={false}
    />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('NotionalInput with warning', () => {
  const component = renderWithTheme(
    <NotionalInput
      notional={notional}
      currencyPairSymbol={currencyPairSymbol}
      updateNotional={updateNotional}
      resetNotional={resetNotional}
      validationMessage={inputValidationMessageWarning}
      showResetButton={false}
      disabled={false}
    />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('NotionalInput with info', () => {
  const component = renderWithTheme(
    <NotionalInput
      notional={notional}
      currencyPairSymbol={currencyPairSymbol}
      updateNotional={updateNotional}
      resetNotional={resetNotional}
      validationMessage={inputValidationMessageInfo}
      showResetButton={false}
      disabled={false}
    />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('NotionalInput disabled', () => {
  const component = renderWithTheme(
    <NotionalInput
      notional={notional}
      currencyPairSymbol={currencyPairSymbol}
      updateNotional={updateNotional}
      resetNotional={resetNotional}
      validationMessage={null}
      showResetButton={false}
      disabled={true}
    />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('NotionalInput show reset button', () => {
  const component = renderWithTheme(
    <NotionalInput
      notional={notional}
      currencyPairSymbol={currencyPairSymbol}
      updateNotional={updateNotional}
      resetNotional={resetNotional}
      validationMessage={null}
      showResetButton={true}
      disabled={false}
    />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

// Light theme version
test('NotionalInput default', () => {
  const component = renderWithTheme(
    <NotionalInput
      notional={notional}
      currencyPairSymbol={currencyPairSymbol}
      updateNotional={updateNotional}
      resetNotional={resetNotional}
      validationMessage={null}
      showResetButton={false}
      disabled={false}
    />,
    'light',
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('NotionalInput with error', () => {
  const component = renderWithTheme(
    <NotionalInput
      notional={notional}
      currencyPairSymbol={currencyPairSymbol}
      updateNotional={updateNotional}
      resetNotional={resetNotional}
      validationMessage={inputValidationMessageError}
      showResetButton={false}
      disabled={false}
    />,
    'light',
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('NotionalInput with warning', () => {
  const component = renderWithTheme(
    <NotionalInput
      notional={notional}
      currencyPairSymbol={currencyPairSymbol}
      updateNotional={updateNotional}
      resetNotional={resetNotional}
      validationMessage={inputValidationMessageWarning}
      showResetButton={false}
      disabled={false}
    />,
    'light',
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('NotionalInput with info', () => {
  const component = renderWithTheme(
    <NotionalInput
      notional={notional}
      currencyPairSymbol={currencyPairSymbol}
      updateNotional={updateNotional}
      resetNotional={resetNotional}
      validationMessage={inputValidationMessageInfo}
      showResetButton={false}
      disabled={false}
    />,
    'light',
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('NotionalInput disabled', () => {
  const component = renderWithTheme(
    <NotionalInput
      notional={notional}
      currencyPairSymbol={currencyPairSymbol}
      updateNotional={updateNotional}
      resetNotional={resetNotional}
      validationMessage={null}
      showResetButton={false}
      disabled={true}
    />,
    'light',
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('NotionalInput show reset button', () => {
  const component = renderWithTheme(
    <NotionalInput
      notional={notional}
      currencyPairSymbol={currencyPairSymbol}
      updateNotional={updateNotional}
      resetNotional={resetNotional}
      validationMessage={null}
      showResetButton={true}
      disabled={false}
    />,
    'light',
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
