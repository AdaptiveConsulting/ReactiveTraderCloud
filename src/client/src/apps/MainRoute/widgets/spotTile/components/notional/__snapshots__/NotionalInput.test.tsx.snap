// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotionalInput default 1`] = `
.c1 {
  grid-area: Currency;
  opacity: 0.59;
  font-size: 0.625rem;
  line-height: 1.2rem;
}

.c0 {
  display: grid;
  grid-template-columns: 30px auto 30px;
  grid-template-areas: 'Currency Input ResetInputValue' '. Message .';
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-rows: 23px 0;
}

.c2 {
  grid-area: Input;
  background: none;
  text-align: center;
  outline: none;
  border: none;
  font-size: 0.75rem;
  width: 80px;
  padding: 2px 0;
  caret-color: #282e39;
  box-shadow: 0px 1px 0px rgba(255,255,255,0.41);
}

.spot-tile:hover .c2:focus,
.c2:focus {
  box-shadow: 0px 1px 0px #5f94f5;
}

<div
  className="c0"
>
  <span
    className="c1"
    data-qa="notional-input__currency-pair-symbol"
  >
    EUR
  </span>
  <input
    className="c2"
    data-qa-id="notional-input__input-eurusd"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value="1,000,000"
  />
</div>
`;

exports[`NotionalInput default 2`] = `
.c1 {
  grid-area: Currency;
  opacity: 0.59;
  font-size: 0.625rem;
  line-height: 1.2rem;
}

.c0 {
  display: grid;
  grid-template-columns: 30px auto 30px;
  grid-template-areas: 'Currency Input ResetInputValue' '. Message .';
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-rows: 23px 0;
}

.c2 {
  grid-area: Input;
  background: none;
  text-align: center;
  outline: none;
  border: none;
  font-size: 0.75rem;
  width: 80px;
  padding: 2px 0;
  caret-color: #fff;
  box-shadow: 0px 1px 0px rgba(40,46,57,0.41);
}

.spot-tile:hover .c2:focus,
.c2:focus {
  box-shadow: 0px 1px 0px #5f94f5;
}

<div
  className="c0"
>
  <span
    className="c1"
    data-qa="notional-input__currency-pair-symbol"
  >
    EUR
  </span>
  <input
    className="c2"
    data-qa-id="notional-input__input-eurusd"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value="1,000,000"
  />
</div>
`;

exports[`NotionalInput disabled 1`] = `
.c1 {
  grid-area: Currency;
  opacity: 0.59;
  font-size: 0.625rem;
  line-height: 1.2rem;
}

.c0 {
  display: grid;
  grid-template-columns: 30px auto 30px;
  grid-template-areas: 'Currency Input ResetInputValue' '. Message .';
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-rows: 23px 0;
}

.c2 {
  grid-area: Input;
  background: none;
  text-align: center;
  outline: none;
  border: none;
  font-size: 0.75rem;
  width: 80px;
  padding: 2px 0;
  caret-color: #282e39;
  opacity: 0.3;
  box-shadow: 0px 1px 0px rgba(255,255,255,0.41);
}

.spot-tile:hover .c2:focus,
.c2:focus {
  box-shadow: 0px 1px 0px #5f94f5;
}

<div
  className="c0"
>
  <span
    className="c1"
    data-qa="notional-input__currency-pair-symbol"
  >
    EUR
  </span>
  <input
    className="c2"
    data-qa-id="notional-input__input-eurusd"
    disabled={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value="1,000,000"
  />
</div>
`;

exports[`NotionalInput disabled 2`] = `
.c1 {
  grid-area: Currency;
  opacity: 0.59;
  font-size: 0.625rem;
  line-height: 1.2rem;
}

.c0 {
  display: grid;
  grid-template-columns: 30px auto 30px;
  grid-template-areas: 'Currency Input ResetInputValue' '. Message .';
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-rows: 23px 0;
}

.c2 {
  grid-area: Input;
  background: none;
  text-align: center;
  outline: none;
  border: none;
  font-size: 0.75rem;
  width: 80px;
  padding: 2px 0;
  caret-color: #fff;
  opacity: 0.3;
  box-shadow: 0px 1px 0px rgba(40,46,57,0.41);
}

.spot-tile:hover .c2:focus,
.c2:focus {
  box-shadow: 0px 1px 0px #5f94f5;
}

<div
  className="c0"
>
  <span
    className="c1"
    data-qa="notional-input__currency-pair-symbol"
  >
    EUR
  </span>
  <input
    className="c2"
    data-qa-id="notional-input__input-eurusd"
    disabled={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value="1,000,000"
  />
</div>
`;

exports[`NotionalInput show reset button 1`] = `
.c1 {
  grid-area: Currency;
  opacity: 0.59;
  font-size: 0.625rem;
  line-height: 1.2rem;
}

.c0 {
  display: grid;
  grid-template-columns: 30px auto 30px;
  grid-template-areas: 'Currency Input ResetInputValue' '. Message .';
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-rows: 23px 0;
}

.c2 {
  grid-area: Input;
  background: none;
  text-align: center;
  outline: none;
  border: none;
  font-size: 0.75rem;
  width: 80px;
  padding: 2px 0;
  caret-color: #282e39;
  box-shadow: 0px 1px 0px rgba(255,255,255,0.41);
}

.spot-tile:hover .c2:focus,
.c2:focus {
  box-shadow: 0px 1px 0px #5f94f5;
}

.c3 {
  background-color: #343a47;
  border: 2px solid #282e39;
  border-radius: 3px;
  margin-left: 8px;
  grid-area: ResetInputValue;
  font-size: 0.625rem;
  line-height: 1.2rem;
}

<div
  className="c0"
>
  <span
    className="c1"
    data-qa="notional-input__currency-pair-symbol"
  >
    EUR
  </span>
  <input
    className="c2"
    data-qa-id="notional-input__input-eurusd"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value="1,000,000"
  />
  <button
    className="c3"
    data-qa="notional-input__reset-input-value"
    onClick={[Function]}
  >
    <i
      className="fas fa-redo fa-flip-horizontal"
    />
  </button>
</div>
`;

exports[`NotionalInput show reset button 2`] = `
.c1 {
  grid-area: Currency;
  opacity: 0.59;
  font-size: 0.625rem;
  line-height: 1.2rem;
}

.c0 {
  display: grid;
  grid-template-columns: 30px auto 30px;
  grid-template-areas: 'Currency Input ResetInputValue' '. Message .';
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-rows: 23px 0;
}

.c2 {
  grid-area: Input;
  background: none;
  text-align: center;
  outline: none;
  border: none;
  font-size: 0.75rem;
  width: 80px;
  padding: 2px 0;
  caret-color: #fff;
  box-shadow: 0px 1px 0px rgba(40,46,57,0.41);
}

.spot-tile:hover .c2:focus,
.c2:focus {
  box-shadow: 0px 1px 0px #5f94f5;
}

.c3 {
  background-color: #fff;
  border: 2px solid #f9f9f9;
  border-radius: 3px;
  margin-left: 8px;
  grid-area: ResetInputValue;
  font-size: 0.625rem;
  line-height: 1.2rem;
}

<div
  className="c0"
>
  <span
    className="c1"
    data-qa="notional-input__currency-pair-symbol"
  >
    EUR
  </span>
  <input
    className="c2"
    data-qa-id="notional-input__input-eurusd"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value="1,000,000"
  />
  <button
    className="c3"
    data-qa="notional-input__reset-input-value"
    onClick={[Function]}
  >
    <i
      className="fas fa-redo fa-flip-horizontal"
    />
  </button>
</div>
`;

exports[`NotionalInput with error 1`] = `
.c1 {
  grid-area: Currency;
  opacity: 0.59;
  font-size: 0.625rem;
  line-height: 1.2rem;
}

.c3 {
  color: #ff274b;
  grid-area: Message;
  font-size: 0.6rem;
  line-height: normal;
  padding-top: 2px;
}

.c0 {
  display: grid;
  grid-template-columns: 30px auto 30px;
  grid-template-areas: 'Currency Input ResetInputValue' '. Message .';
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-rows: 23px 13px;
  margin-bottom: -0.5rem;
}

.c2 {
  grid-area: Input;
  background: none;
  text-align: center;
  outline: none;
  border: none;
  font-size: 0.75rem;
  width: 80px;
  padding: 2px 0;
  caret-color: #282e39;
  box-shadow: 0px 1px 0px #ff274b;
}

<div
  className="c0"
>
  <span
    className="c1"
    data-qa="notional-input__currency-pair-symbol"
  >
    EUR
  </span>
  <input
    className="c2"
    data-qa-id="notional-input__input-eurusd"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value="1,000,000"
  />
  <div
    className="c3"
    data-qa="notional-input__validation-message"
  >
    Some error
  </div>
</div>
`;

exports[`NotionalInput with error 2`] = `
.c1 {
  grid-area: Currency;
  opacity: 0.59;
  font-size: 0.625rem;
  line-height: 1.2rem;
}

.c3 {
  color: #ff274b;
  grid-area: Message;
  font-size: 0.6rem;
  line-height: normal;
  padding-top: 2px;
}

.c0 {
  display: grid;
  grid-template-columns: 30px auto 30px;
  grid-template-areas: 'Currency Input ResetInputValue' '. Message .';
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-rows: 23px 13px;
  margin-bottom: -0.5rem;
}

.c2 {
  grid-area: Input;
  background: none;
  text-align: center;
  outline: none;
  border: none;
  font-size: 0.75rem;
  width: 80px;
  padding: 2px 0;
  caret-color: #fff;
  box-shadow: 0px 1px 0px #ff274b;
}

<div
  className="c0"
>
  <span
    className="c1"
    data-qa="notional-input__currency-pair-symbol"
  >
    EUR
  </span>
  <input
    className="c2"
    data-qa-id="notional-input__input-eurusd"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value="1,000,000"
  />
  <div
    className="c3"
    data-qa="notional-input__validation-message"
  >
    Some error
  </div>
</div>
`;

exports[`NotionalInput with info 1`] = `
.c1 {
  grid-area: Currency;
  opacity: 0.59;
  font-size: 0.625rem;
  line-height: 1.2rem;
}

.c3 {
  color: #5f94f5;
  grid-area: Message;
  font-size: 0.6rem;
  line-height: normal;
  padding-top: 2px;
}

.c0 {
  display: grid;
  grid-template-columns: 30px auto 30px;
  grid-template-areas: 'Currency Input ResetInputValue' '. Message .';
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-rows: 23px 13px;
  margin-bottom: -0.5rem;
}

.c2 {
  grid-area: Input;
  background: none;
  text-align: center;
  outline: none;
  border: none;
  font-size: 0.75rem;
  width: 80px;
  padding: 2px 0;
  caret-color: #282e39;
  box-shadow: 0px 1px 0px #5f94f5;
}

<div
  className="c0"
>
  <span
    className="c1"
    data-qa="notional-input__currency-pair-symbol"
  >
    EUR
  </span>
  <input
    className="c2"
    data-qa-id="notional-input__input-eurusd"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value="1,000,000"
  />
  <div
    className="c3"
    data-qa="notional-input__validation-message"
  >
    Some information
  </div>
</div>
`;

exports[`NotionalInput with info 2`] = `
.c1 {
  grid-area: Currency;
  opacity: 0.59;
  font-size: 0.625rem;
  line-height: 1.2rem;
}

.c3 {
  color: #5f94f5;
  grid-area: Message;
  font-size: 0.6rem;
  line-height: normal;
  padding-top: 2px;
}

.c0 {
  display: grid;
  grid-template-columns: 30px auto 30px;
  grid-template-areas: 'Currency Input ResetInputValue' '. Message .';
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-rows: 23px 13px;
  margin-bottom: -0.5rem;
}

.c2 {
  grid-area: Input;
  background: none;
  text-align: center;
  outline: none;
  border: none;
  font-size: 0.75rem;
  width: 80px;
  padding: 2px 0;
  caret-color: #fff;
  box-shadow: 0px 1px 0px #5f94f5;
}

<div
  className="c0"
>
  <span
    className="c1"
    data-qa="notional-input__currency-pair-symbol"
  >
    EUR
  </span>
  <input
    className="c2"
    data-qa-id="notional-input__input-eurusd"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value="1,000,000"
  />
  <div
    className="c3"
    data-qa="notional-input__validation-message"
  >
    Some information
  </div>
</div>
`;

exports[`NotionalInput with warning 1`] = `
.c1 {
  grid-area: Currency;
  opacity: 0.59;
  font-size: 0.625rem;
  line-height: 1.2rem;
}

.c3 {
  color: #ff8d00;
  grid-area: Message;
  font-size: 0.6rem;
  line-height: normal;
  padding-top: 2px;
}

.c0 {
  display: grid;
  grid-template-columns: 30px auto 30px;
  grid-template-areas: 'Currency Input ResetInputValue' '. Message .';
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-rows: 23px 13px;
  margin-bottom: -0.5rem;
}

.c2 {
  grid-area: Input;
  background: none;
  text-align: center;
  outline: none;
  border: none;
  font-size: 0.75rem;
  width: 80px;
  padding: 2px 0;
  caret-color: #282e39;
  box-shadow: 0px 1px 0px #ff8d00;
}

<div
  className="c0"
>
  <span
    className="c1"
    data-qa="notional-input__currency-pair-symbol"
  >
    EUR
  </span>
  <input
    className="c2"
    data-qa-id="notional-input__input-eurusd"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value="1,000,000"
  />
  <div
    className="c3"
    data-qa="notional-input__validation-message"
  >
    Some warning
  </div>
</div>
`;

exports[`NotionalInput with warning 2`] = `
.c1 {
  grid-area: Currency;
  opacity: 0.59;
  font-size: 0.625rem;
  line-height: 1.2rem;
}

.c3 {
  color: #ff8d00;
  grid-area: Message;
  font-size: 0.6rem;
  line-height: normal;
  padding-top: 2px;
}

.c0 {
  display: grid;
  grid-template-columns: 30px auto 30px;
  grid-template-areas: 'Currency Input ResetInputValue' '. Message .';
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-rows: 23px 13px;
  margin-bottom: -0.5rem;
}

.c2 {
  grid-area: Input;
  background: none;
  text-align: center;
  outline: none;
  border: none;
  font-size: 0.75rem;
  width: 80px;
  padding: 2px 0;
  caret-color: #fff;
  box-shadow: 0px 1px 0px #ff8d00;
}

<div
  className="c0"
>
  <span
    className="c1"
    data-qa="notional-input__currency-pair-symbol"
  >
    EUR
  </span>
  <input
    className="c2"
    data-qa-id="notional-input__input-eurusd"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value="1,000,000"
  />
  <div
    className="c3"
    data-qa="notional-input__validation-message"
  >
    Some warning
  </div>
</div>
`;
